include/pybind11/pytypes.h

```diff
@@ -1059,6 +1059,35 @@ inline str::str(const bytes& b) {
     m_ptr = obj.release().ptr();
 }
 
+/// \addtogroup pytypes
+/// @{
+class bytearray : public object {
+public:
+    PYBIND11_OBJECT(bytearray, object, PyByteArray_Check)
```

Comment:
What about `PYBIND11_OBJECT_CVT(bytearray, object, PyByteArray_Check, PyByteArray_FromObject)`? This would allow to call `py::bytearray(some_other_object_with_buffer_protocol)`.

Reply:
Done.